### PR TITLE
PWN-3566 - switch to Main tab or back when Send is finished

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3566

## Description of Work

- return to previous screen when send is finished
- or switch to Main tab

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/send/ui/main/SendFragment.kt [hunk 1]
```diff
@@ -124,16 +122,22 @@ class SendFragment :
...
+        childFragmentManager.setFragmentResultListener(KEY_REQUEST_SEND, viewLifecycleOwner) { _, result ->
+            if (result.containsKey(EXTRA_RESULT_KEY_DISMISS)) closeScreenAndReturn()
+        }
+
         presenter.loadInitialData()
         checkClipBoard()
     }
 
+    private fun closeScreenAndReturn() {
+        if (backStackEntryCount() > 1) popBackStack() else replaceFragment(HomeFragment.create())
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/main/SendFragment.kt:138]: is it ok to create HomeFragment here from scratch? maybe i can use bottom tabs to switch to Home??

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/send/ui/main/SendFragment.kt (hunk 1 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/send/ui/main/SendFragment.kt:138]: I think we already have an ability to switch between tabs if you call `popBackStack()`

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/send/ui/main/SendFragment.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/send/ui/main/SendFragment.kt:138]: popBackStack closes the app itself. I think we are replacing fragments when switching tabs

Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/send/ui/main/SendFragment.kt:138]: Look into `MainFragment`, there is already implemented switching between tabs when onBackPressed
```
override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requireActivity().onBackPressedDispatcher.addCallback(this) {
            if (binding.bottomNavigation.selectedItemId != R.id.itemHome) {
                navigate(R.id.itemHome)
                binding.bottomNavigation.menu[0].isChecked = true
            } else {
                requireActivity().finish()
            }
        }
    }
```

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/send/ui/main/SendFragment.kt]: +++

**Review by kuznetsov — APPROVED** [review]

**Review by davran1001 — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

**Pull request merged**